### DISCUSSION TITLE
Rendering Grease Pencil while using screen passes

### DISCUSSION THREAD

**Discussion by Dereuz24** [bnpr/Malt#588] [Q&A]
Hi there,

I was setting up the Bloom effect of the example file again in another file from scratch and found that it doesn't render Grease Pencil objects. However these GP objects do appear on the render when I move them to the sample file. 

Everything seems to be looking exactly the same in both Bloom shaders, the effect works and outputs in the rendered video, default render and render layer also have the same structure so I'm having difficulty understanding what could be obfuscating the GP rendering engine when I recreate the bloom effect with the same ordering of the layers. 

Isn't it supposed to show up always on top of everything or are there any settings that could cause the GP to be pushed behind the objects rendered with Malt?

The GP object also doesn't render when I remove the Bloom Screen Pass node completely from the default render. I'm scratching my head over the kind of setting that would cause this independently of the GP configuration (because this same object can be rendered in the sample bloom blender file, on eevee on the current file, etc)
![1](https://github.com/user-attachments/assets/d4604193-9ae5-4599-8223-caaccd89ef34)
![2](https://github.com/user-attachments/assets/ae6f54b7-1afa-480e-bff1-28f489d0da62)
![3](https://github.com/user-attachments/assets/2c20cf1b-6427-4bac-b91c-91b5c39b3215)

**Comment by Dereuz24** [comment]
A second question I need help with. I can't seem to figure this out but the ambient occlusion node gives me a different result in render than in viewport. Even when I crack the samples to 5000. I understand is all done on screen-space but the radius of the effect is radically different in render compared to what I see on screen, sometimes like 10 times smaller

Is there a setting I'm missing here or I just have to go back and fort by eye comparing to a final render instead of the viewport?

EDIT: I just tested on a heavy scene, it's like 20 times difference

**Reply by pragma37** [reply]
I can't repro it. It looks roughly the same:
![imagen](https://github.com/user-attachments/assets/d2262bde-d1fa-4f3f-b2ff-ff19cf3af0d3)

Do you maybe have some parameter override?
https://malt3d.com/Documentation/Settings/#parameter-overrides

**Comment by Dereuz24** [comment]
Apologizes to bother you again while you grinding at the NPR prototype @pragma37 but do you have a rough idea on why would the Grease pencil not render with these passes? 

It seems to work in your original bloom example file but after reproducing the effect it's just missing from the renders, so I know there's something I must be missing but I have no idea what it could be

**Reply by pragma37** [reply]
Grease Pencil uses its own renderer that Malt has no control over.
The only thing that can affect it from the Malt side is the Depth output (and only in final renders, not in the viewport), since GPencil uses that for compositing.
Do you maybe have a custom `Depth` output plugged into the `Render Output` node?
By default your render nodes should look like this:
![imagen](https://github.com/user-attachments/assets/9e95a54f-f4a5-4573-b211-caaa5aac128c)
